android/app/src/main/java/com/happy/friendogly/data/repository/ChatMessageRepositoryImpl.kt

```diff
@@ -0,0 +1,36 @@
+package com.happy.friendogly.data.repository
+
+import com.happy.friendogly.data.mapper.toDomain
+import com.happy.friendogly.data.source.ChatMessageDataSource
+import com.happy.friendogly.domain.model.ChatComponent
+import com.happy.friendogly.domain.repository.ChatMessageRepository
+import com.happy.friendogly.local.mapper.toData
+import com.happy.friendogly.local.room.ChatRoomDao
+import java.time.LocalDateTime
+import javax.inject.Inject
+
+class ChatMessageRepositoryImpl
+    @Inject
+    constructor(
+        private val dataSource: ChatMessageDataSource,
+        private val chatRoomDao: ChatRoomDao,
+    ) : ChatMessageRepository {
+        override suspend fun getAllChatMessages(
+            myMemberId: Long,
+            chatRoomId: Long,
+        ): Result<List<ChatComponent>> {
+            val chatMessages = dataSource.getAllChatMessages(chatRoomId).mapCatching { it.toDomain(myMemberId) }
+            chatRoomDao.addMessagesToChatRoom(chatRoomId, chatMessages.getOrThrow().toData())
+            return chatMessages
+        }
+
+        override suspend fun getChatMessageByTime(
+            myMemberId: Long,
+            chatRoomId: Long,
+            since: LocalDateTime,
+            until: LocalDateTime,
```

Comment:
since, until 이 무엇을 의미하는 건가요?!

Reply:
`특정 날짜+시간(since) ~ 특정 날짜+시간(until)`사이에 온 채팅메시지만 받고 싶다! 라는 의미입니다!
모든 채팅 메시지를 받기에는 데이터의 양이 너무 많아서, 나눠서 받기 위해 사용할 예정입니다~

이때 개발했을 때는, 아직 서버와 협의중인 부분이 있었기 때문에 메서드를 사용하지는 않았습니다 🥲

Reply:
지금은 페이징을 적용했기 때문에 사용중입니다~